### PR TITLE
fix: include appropriate oidcs scopes + misc refactor

### PR DESCRIPTION
**Pull request by ddlees**
## Description

*Describe your changes in detail*

When testing against Auth0 we discovered that the id_token was not being returned as expected. To remedy this we explicitly declare all the requires scopes in this PR.

## Motivation and Context

This PR addresses: BED-4768

*Why is this change required? What problem does it solve?*

## How Has This Been Tested?

*Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc.*

I have updated all the tests to account for my change.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

**Review by mistahj67 — COMMENTED** [review]
These worked on authentik locally for me. Did we want to add the OIDC email_verified check and / or fallback with name for user lookup?

File: cmd/api/src/api/auth.go [hunk 1]
```diff
@@ -329,30 +329,35 @@ func SetSecureBrowserCookie(request *http.Request, response http.ResponseWriter,
...
+	case model.SessionAuthProviderOIDC:
 		auditLogFields["auth_type"] = auth.ProviderTypeOIDC
-		auditLogFields["oidc_provider_id"] = typedAuthProvider.ID
-		auditLogFields["sso_provider_id"] = typedAuthProvider.SSOProviderID
+		if ssoProvider.OIDCProvider != nil {
+			auditLogFields["oidc_provider_id"] = ssoProvider.OIDCProvider.ID
+			authProvider = *ssoProvider.OIDCProvider
+		}
+    DeleteBrowserCookie(request, response, AuthPKCECookieName)
+    DeleteBrowserCookie(request, response, AuthPKCECookieName)
```
Comment (mvlipka) [inline cmd/api/src/api/auth.go:360]: Copy/paste miss perhaps?
```
DeleteBrowserCookie(request, response, AuthStateCookieName)
DeleteBrowserCookie(request, response, AuthPKCECookieName)

```

Comment (ddlees) [reply cmd/api/src/api/auth.go:360]: doh!

File: cmd/api/src/api/saml/saml.go [hunk 2]
```diff
@@ -348,7 +349,13 @@ func (s ProviderResource) serveAssertionConsumerService(response http.ResponseWr
 			log.Errorf("[SAML] Failed to lookup user for SAML provider %s: %v", s.serviceProvider.Config.Name, err)
 			s.writeAPIErrorResponse(request, response, http.StatusBadRequest, "session assertion does not meet the requirements for user lookup")
 		} else {
-			s.authenticator.CreateSSOSession(request, response, principalName, s.serviceProvider.Config)
+			s.authenticator.CreateSSOSession(request, response, principalName, model.SSOProvider{
+				Type: model.SessionAuthProviderSAML,
+				Name: s.serviceProvider.Config.Name,
+				Slug: s.serviceProvider.Config.Name,
```
Comment (mvlipka) [inline cmd/api/src/api/saml/saml.go:355]: It doesn't _look_ like this gets used, but, there is the potential that the `Slug` would be different than the `Name` in the database.
It looks like it should be fine here,  but wanted to call that out.

**Review by mvlipka — APPROVED** [review]
LGTM :shipit:

**Pull request merged**